packages/core/lib/test.js

```diff
@@ -116,27 +116,14 @@ const Test = {
 
     await this.performInitialDeploy(config, testResolver);
 
-    const solidityCompilationOutput = compilations.reduce(
-      (a, compilation) => {
-        if (compilation.compiler && compilation.compiler.name === "solc") {
-          a.sourceIndexes = a.sourceIndexes.concat(compilation.sourceIndexes);
-          a.contracts = a.contracts.concat(compilation.contracts);
-        }
-        return a;
-      },
-      { sourceIndexes: [], contracts: [] }
+    const sourcePaths = [].concat(
+      ...compilations.map(compilation => compilation.sourceIndexes) //we don't need the indices here, just the paths
```

Comment:
Hm, are we definitely saying, then, that this `sourceIndexes` field explicitly excludes internal sources?

Can you add a comment in the corresponding @truffle/compile-common types definition, to that effect? We'll probably need to add a new `internalSourceIndexes` property...

Reply:
It never occurred to me that it might include internal sources, but I assume it has to exclude them, as including them would take extra work that I have to assume was not done (nor has their inclusion caused any problems thus far, which it presumably would have).

I don't see why we need an `internalSourceIndexes` property.  Internal source indices are handled in the internal source objects.

I can add that comment, though.

Thought: One potential problem that might come up with internal sources is that the user sources might not be consecutive.  This could potentially cause problems in `SolidityTest.js` if it results in a call to `resolve(undefined)`.  Do you know if that causes an error?  If you don't know that it doesn't, I'll go add a guard around that to be sure.

Reply:
It seems like it might cause an error - adding that guard seems appropriate

Reply:
I decided to add a filter instead, heh.  Same effect.